samples/embedded-auth-with-sdk/src/main/resources/templates/register.html

```diff
@@ -64,15 +64,17 @@ <h2 class="text-center">Create Account</h2>
         <div th:if="${errors}" class="alert alert-danger">
             <span th:text="${errors}"></span>
         </div>
-        <div class="form-group">
-            <input type="text" class="form-control" name="firstname" placeholder="First Name" required="required">
-        </div>
-        <div class="form-group">
-            <input type="text" class="form-control" name="lastname" placeholder="Last Name" required="required">
-        </div>
-        <div class="form-group">
-            <input type="text" class="form-control" name="email" placeholder="Email" required="required">
+
+        <!--display the user attributes for register form dynamically-->
+        <div th:if="${not #lists.isEmpty(userProfileAttributes)}" class="container" th:align="center">
+            <th:block th:each="userProfileAttribute: ${userProfileAttributes}">
+                <div class="form-group">
+                    <input type="text" class="form-control" name="userProfileAttribute[]" th:placeholder="${userProfileAttribute.getLabel()}" th:required="${userProfileAttribute.required}">
+                </div>
```

Comment:
Pretty neat that the `required` attribute returned from the user profile remediation can be used to enforce client side `required` constraint

Reply:
Looks good to me